### PR TITLE
Add "About" section to settings page

### PR DESCRIPTION
**Pull request by GeorgeHoneywood**
Resolves https://github.com/Zverik/every_door/issues/238

This adds some meta info to the settings page, in a new "About" section, as per the screenshot. Also moves the system log to it's own dedicated button.

I did try a dedicated "About" page, but it seemed a bit sparse, so a section on the main settings page seemed better.

As always, thanks!

### CODE REVIEW

**Comment by Zverik** [comment]
Thanks for the PR, George. I think the About page needs to be its own page, because there are so many things. Right now it double the number of entries on the Settings page. And it needs to be structured better:

* Title and version number
* Log and report
* Links
* FAQ excerpt

It's not easy, hence I postponed working on the page :)

**Comment by GeorgeHoneywood** [comment]
Thanks for the feedback!

I did a bit of a redesign — moved it to a separate page, using the order of your bullet points. I think this version makes a good amount of sense, but one thing I'm not sure about is the FAQs. The way I have done it for the screenshot is by copying the markdown from the website repo into a multiline string, then rendering it with [`flutter_markdown`](https://pub.dev/packages/flutter_markdown). This obviously isn't great as it is manual, and doesn't internationalise. I suppose we can copy strings from the website into the translations and do them through weblate, but having two independent copies of the FAQ doesn't seem great either.



If this is something it'd be easier you you to do yourself, just let me know. If not I can make changes/fixup required :)

P.S. I realise maybe a separate attribution page for licences of dependencies and other things is needed, but this would be a different thing.

**Comment by Zverik** [comment]
Hey, this looks much better :) Actually I'm fine with everything except the FAQ.

I'd like the FAQ sections to be expandable and collapsed by default. The "German" entry is obsolete, and the last one I'll move to the readme here — no use reading that in the app. So basically this is a technical task to make these collapsed, and I'll rewrite the content (and make it translatable) after we merge this.

The idea is to remove FAQ from the website entirely, so it would serve only as a downloading point.

**Comment by GeorgeHoneywood** [comment]
I've made the sections collapsible using `ExpansionTile`, seems to work well.



It is using [`flutter_markdown`](https://pub.dev/packages/flutter_markdown) so that we can have links, monospace and bullet points without having to directly write flutter rich text.

> I'll rewrite the content (and make it translatable) after we merge this.

Sounds good :) Thanks again for having reviewing this!

File: lib/l10n/app_en.arb [hunk 1]
```diff
@@ -264,6 +264,100 @@
   "@settingsDefaultPayment": {
     "description": "Settings switch for choosing default payment types, e.g. visa and mastercard."
   },
+  "settingsSystem": "System",
+  "@settingsSystem": {
+    "description": "Settings section for System settings."
+  },
+  "settingsUseGooglePositioning": "Use Google-enhanced Positioning",
```
[the remark this answers is not in the document]
Comment (GeorgeHoneywood) [reply lib/l10n/app_en.arb]: Fixed :)

File: lib/l10n/app_en.arb [hunk 2]
```diff
@@ -264,6 +264,100 @@
...
+  "@settingsUseGooglePositioning": {
+    "description": "Settings switch to enable Google enhanced geolocation."
+  },
+  "settingsAbout": "About",
+  "@settingsAbout": {
+    "description": "Title for the About section in settings."
+  },
+  "aboutVersion": "About {appTitle} v{appVersion}",
```
[the remark this answers is not in the document]
Comment (Zverik) [reply lib/l10n/app_en.arb]: I think "Help improve {appname}" should stay, because word ordering might be different in some languages. Others are okay.

**Review by Zverik — APPROVED** [review]
Thank you!

**Pull request merged**